ros_buildfarm/status_page.py

```diff
@@ -530,6 +531,298 @@ def _get_comparable_loose_versions(version_str1, version_str2):
+        return '<td id=".{0}"><div>{1}</div></td>'.format(name, data)
+    else:
+        return '<td><div>{0}</div></td>'.format(data)
+
+
+def _filter_tag_wrap(label):
+    return '<span class="ht">_{0}</span>'.format(label)
+
+
+def _name_query_wrap(name):
+    return '<a href="?q=.{0}">{0}</a>'.format(name)
+
+
+def _format_repo_table_row(name, data):
+    # convert data in dictionary into data that can be used as an html table row
+    row = {}
+
+    repos_blocking = data.get('repos_blocking', [])
+    repos_blocked_by = data.get('repos_blocked_by', {})
+
+    # tags for filtering
+    tags = ''
+    if data['released']:
+        tags += _filter_tag_wrap('RELEASED')
+    else:
+        tags += _filter_tag_wrap('UNRELEASED')
+
+        if len(repos_blocked_by) > 0:
+            tags += _filter_tag_wrap('BLOCKED')
+        elif len(repos_blocking) > 0:
+            tags += _filter_tag_wrap('UNBLOCKED_BLOCKING')
+        else:
+            tags += _filter_tag_wrap('UNBLOCKED_UNBLOCKING')
+
+    url = data.get('url', None)
+    if url:
+        name_html = '<a href="{0}">{1}</a>'.format(url, name)
+    else:
+        name_html = name
+    name_cell = name_html + tags
+
+    # repo name cell
+    row['name'] = _table_cell_wrap(name_cell, name)
+
+    # whether or not the repo has been released
+    row['version'] = _table_cell_wrap(data.get('version', 'Not released'))
+
+    # num repos blocked by
+    row['num_repos_blocked_by'] = _table_cell_wrap(len(repos_blocked_by))
+
+    # repos blocked by
+    row['repos_blocked_by'] = _table_cell_wrap('<br />'.join(
+        _name_query_wrap(repo) for repo in sorted(repos_blocked_by.keys())))
+
+    # maintainers info
+    maintainers = data.get('maintainers', {})
+    maintainers_cell = ''
+    for repo_name in sorted(maintainers.keys()):
+        maintainers_cell += repo_name + ':<br />'
+        maintainers_cell += '<br />'.join(
+            '&emsp;<a href="mailto:{0}">{1}</a>'
+            .format(maintainers[repo_name][name], name)
+            for name in sorted(maintainers[repo_name].keys()))
+        maintainers_cell += '<br />'
+    row['maintainers_of_repos_blocked_by'] = _table_cell_wrap(maintainers_cell)
+
+    # num repos blocking recursively
+    row['num_repos_recursively_blocked'] = _table_cell_wrap(
+        len(data.get('recursive_repos_blocking', [])))
+
+    # num repos blocking
+    row['num_repos_blocked'] = _table_cell_wrap(len(repos_blocking))
+
+    # repos blocking by not being released
+    row['repos_blocked'] = _table_cell_wrap('<br />'.join(
+        _name_query_wrap(repo) for repo in sorted(repos_blocking)))
+
+    return row
+
+
+def _is_released(repo, dist_file):
+    return repo in dist_file.repositories and \
+        dist_file.repositories[repo].release_repository is not None and \
+        dist_file.repositories[repo].release_repository.version is not None
+
+
+def _get_blocked_releases_info(
+        config_url, rosdistro_name, repo_names=None, depth=1):
```

Comment:
@scpeters do you see what you expect for libccd now? http://54.183.65.232/status_page/blocked_releases/blocked_releases_kinetic.html?q=fcl